draft-ietf-ppm-dap.md

```diff
@@ -621,6 +621,9 @@ number generator. Each task has the following parameters associated with it:
   task must have.
 * `max_batch_lifetime`: The maximum number of times a batch of reports may be
   queried by the Collector.
+* `max_task_lifetime`: The timestamp up to which clients are allowed to upload
```

Comment:
I think defining this as a timestamp (i.e., the encoding of an instant in time that might appear in a message) is wrong. The `max_task_lifetime` is an instant in time.

`lifetime` also seems like the wrong word. For one thing, it collides awkwardly with `max_batch_lifetime` (which refers to a totally different quantity). For another, "lifetime" implies to me a span of time, or a duration, but this is a single instant. Would `task_expiration` be better?

Reply:
I agree `task_expiration` is a better name if it's defined as a timestamp.

What's the main concern of encoding an instant in time in the protocol message (sorry if this is  obvious). Defining it as a duration will make it very hard for client to check this, since a client doesn't know the starting time of a task.

Reply:
I think the distinction is that `task_expiration` should be defined as an abstract instant in time, not a timestamp representing that instant. Since the expiration is going to be shared out of bounds, and won't appear in encoded form anywhere in the protocol, it would be cleanest to define it as a point in time, and not define a timestamp encoding for it that the protocol doesn't use.